src/mathjax/math.ts

```diff
@@ -0,0 +1,206 @@
+import marpitPlugin from '@marp-team/marpit/plugin'
+
+import { mathjax } from 'mathjax-full/js/mathjax'
+import { TeX } from 'mathjax-full/js/input/tex'
+import { SVG } from 'mathjax-full/js/output/svg'
+import { liteAdaptor } from 'mathjax-full/js/adaptors/liteAdaptor'
+import { RegisterHTMLHandler } from 'mathjax-full/js/handlers/html'
+import { AllPackages } from 'mathjax-full/js/input/tex/AllPackages'
+
+const adaptor = liteAdaptor()
+RegisterHTMLHandler(adaptor)
+const tex = new TeX({ packages: AllPackages })
+const svg = new SVG({ fontCache: 'none' })
+const mathDocument = mathjax.document('', { InputJax: tex, OutputJax: svg })
+
+interface MathOptionsInterface {
+  katexOption?: object
+}
+
+export type MathOptions = boolean | MathOptionsInterface
+
+/**
+ * marp-core math plugin
+ *
+ * It is implemented based on markdown-it-katex plugin. However, that is no
+ * longer maintained by author. So we have ported math typesetting parser.
+ *
+ * @see https://github.com/waylonflinn/markdown-it-katex
+ */
+export const markdown = marpitPlugin(
+  (md, updateState: (rendered: boolean) => void = () => {}) => {
+    const genOpts = (display: boolean) => {
+      const math: MathOptions = md.marpit.options.math
+
+      return {
+        ...(typeof math === 'object' && typeof math.katexOption === 'object'
+          ? math.katexOption
+          : {}),
+        display,
+      }
+    }
+
+    md.core.ruler.before('block', 'marp_math_initialize', (state) => {
+      if (state.inlineMode) return
+
+      updateState(false)
+
+      if (md.marpit.options.math) {
+        md.block.ruler.enable('marp_math_block')
+        md.inline.ruler.enable('marp_math_inline')
+      } else {
+        md.block.ruler.disable('marp_math_block')
+        md.inline.ruler.disable('marp_math_inline')
+      }
+    })
+
+    // Inline
+    md.inline.ruler.after('escape', 'marp_math_inline', (state, silent) => {
+      if (parseInlineMath(state, silent)) {
+        updateState(true)
+        return true
+      }
+      return false
+    })
+
+    md.renderer.rules.marp_math_inline = (tokens, idx) => {
+      const { content } = tokens[idx]
+
+      try {
+        return adaptor.outerHTML(mathDocument.convert(content, genOpts(false)))
+      } catch (e) {
+        console.warn(e)
+        return content
+      }
+    }
+
+    // Block
+    md.block.ruler.after(
+      'blockquote',
+      'marp_math_block',
+      (state, start, end, silent) => {
+        if (parseMathBlock(state, start, end, silent)) {
+          updateState(true)
+          return true
+        }
+        return false
+      },
+      { alt: ['paragraph', 'reference', 'blockquote', 'list'] }
+    )
+
+    md.renderer.rules.marp_math_block = (tokens, idx) => {
+      const { content } = tokens[idx]
+
+      try {
+        return `<p>${adaptor.outerHTML(
+          mathDocument.convert(content, genOpts(true))
```

Comment:
By comparing to KaTeX rendering, the rendered math block is not centering.

Reply:
Solved, check it.

```js
const { Marp } = require("./lib/marp.js")
const marp = new Marp({ math: true, mathjax: false})

const md = `
# Elementary Math

$$
1+1 = 2
$$
`
const r = marp.render(md)
console.log(r.html + `<style>${r.css}</style>`)
```